main.go

```diff
@@ -268,37 +269,46 @@ func main() {
 			return perrors.Wrapf(err, "unable to create remote cluster cache tracker")
 		}
 
-		// Check for non-supervisor VSphereCluster and start controller if found
-		gvr := infrav1.GroupVersion.WithResource(reflect.TypeOf(&infrav1.VSphereCluster{}).Elem().Name())
-		isNonSupervisorCRDLoaded, err := isCRDDeployed(mgr, gvr)
-		if err != nil {
-			return err
+		govmomiGVR := infrav1.GroupVersion.WithResource(reflect.TypeOf(&infrav1.VSphereCluster{}).Elem().Name())
+		supervisorGVR := vmwarev1.GroupVersion.WithResource(reflect.TypeOf(&vmwarev1.VSphereCluster{}).Elem().Name())
+
+		var isSupervisorCRDLoaded, isGovmoniCRDLoaded bool
+		if err := wait.PollUntilContextTimeout(ctx, 5*time.Second, 30*time.Second, true, func(_ context.Context) (bool, error) {
+			var errGovmomi, errSupervisor error
+
+			// Check for non-supervisor VSphereCluster and start controller if found
+			isGovmoniCRDLoaded, errGovmomi = isCRDDeployed(mgr, govmomiGVR)
+
+			// Check for supervisor VSphereCluster and start controller if found
+			isSupervisorCRDLoaded, errSupervisor = isCRDDeployed(mgr, supervisorGVR)
+
+			if (isGovmoniCRDLoaded && errGovmomi == nil) || (isSupervisorCRDLoaded && errSupervisor == nil) {
+				return true, nil
+			}
+			return false, nil
+		}); err != nil {
+			return fmt.Errorf("failed to detect CRDs: %w", err)
+		}
+
+		// Continuing startup does not make sense without having managers added.
+		if !isSupervisorCRDLoaded && !isGovmoniCRDLoaded {
+			return errors.New("neither supervisor nor govmomi CRDs detected")
 		}
```

Comment:
This is not possible anymore. One of them has to be true to not end up in an error in l.289